go/parquet/metadata/file.go

```diff
@@ -0,0 +1,478 @@
+func (f *FileMetaDataBuilder) GetFileCryptoMetaData() *FileCryptoMetadata {
+	if f.cryptoMetadata == nil {
+		return nil
+	}
+
+	props := f.props.FileEncryptionProperties()
+	f.cryptoMetadata.EncryptionAlgorithm = props.Algorithm().ToThrift()
+	keyMetadata := props.FooterKeyMetadata()
+	if keyMetadata != "" {
+		f.cryptoMetadata.KeyMetadata = []byte(keyMetadata)
+	}
+
+	return &FileCryptoMetadata{f.cryptoMetadata, 0}
+}
+
+// AppendRowGroup adds a rowgroup to the list and returns a builder
+// for that row group
+func (f *FileMetaDataBuilder) AppendRowGroup() *RowGroupMetaDataBuilder {
+	if f.rowGroups == nil {
+		f.rowGroups = make([]*format.RowGroup, 0, 1)
+	}
+
+	rg := format.NewRowGroup()
+	f.rowGroups = append(f.rowGroups, rg)
+	f.currentRgBldr = NewRowGroupMetaDataBuilder(f.props, f.schema, rg)
+	return f.currentRgBldr
+}
+
+// Finish will finalize the metadata of the number of rows, row groups,
+// version etc. This will clear out this filemetadatabuilder so it can
+// be re-used
+func (f *FileMetaDataBuilder) Finish() (*FileMetaData, error) {
+	totalRows := int64(0)
+	for _, rg := range f.rowGroups {
+		totalRows += rg.NumRows
+	}
+	f.metadata.NumRows = totalRows
+	f.metadata.RowGroups = f.rowGroups
+	switch f.props.Version() {
+	case parquet.V1:
+		f.metadata.Version = 1
+	case parquet.V2:
+		f.metadata.Version = 2
+	default:
+		f.metadata.Version = 0
+	}
+	createdBy := f.props.CreatedBy()
+	f.metadata.CreatedBy = &createdBy
+
+	// Users cannot set the `ColumnOrder` since we donot not have user defined sort order
+	// in the spec yet.
+	// We always default to `TYPE_DEFINED_ORDER`. We can expose it in
+	// the API once we have user defined sort orders in the Parquet format.
+	// TypeDefinedOrder implies choose SortOrder based on ConvertedType/PhysicalType
+	typeDefined := format.NewTypeDefinedOrder()
+	colOrder := &format.ColumnOrder{TYPE_ORDER: typeDefined}
+	f.metadata.ColumnOrders = make([]*format.ColumnOrder, f.schema.NumColumns())
+	for idx := range f.metadata.ColumnOrders {
+		f.metadata.ColumnOrders[idx] = colOrder
+	}
+
+	fileEncProps := f.props.FileEncryptionProperties()
+	if fileEncProps != nil && !fileEncProps.EncryptedFooter() {
+		var signingAlgo parquet.Algorithm
+		algo := fileEncProps.Algorithm()
+		signingAlgo.Aad.AadFileUnique = algo.Aad.AadFileUnique
+		signingAlgo.Aad.SupplyAadPrefix = algo.Aad.SupplyAadPrefix
+		if !algo.Aad.SupplyAadPrefix {
+			signingAlgo.Aad.AadPrefix = algo.Aad.AadPrefix
+		}
+		signingAlgo.Algo = parquet.AesGcm
+		f.metadata.EncryptionAlgorithm = signingAlgo.ToThrift()
+		footerSigningMetadata := f.props.FileEncryptionProperties().FooterKeyMetadata()
+		if footerSigningMetadata != "" {
+			f.metadata.FooterSigningKeyMetadata = []byte(footerSigningMetadata)
+		}
+	}
+
+	f.metadata.Schema = schema.ToThrift(f.schema.Root())
+	f.metadata.KeyValueMetadata = f.kvmeta
+
+	out := &FileMetaData{
+		FileMetaData: f.metadata,
+		version:      NewAppVersion(f.metadata.GetCreatedBy()),
```

Comment:
just want to make sure this references go someplace by default?

Reply:
at the top of `go/parquet/writer_properties.go` there is a constant `DefaultCreatedBy` which references `parquet-go`